### PR TITLE
feat: add github.updateIssueComment component

### LINKED ISSUE #5361: [GitHub] Add github.updateIssueComment component
Add a new GitHub action component to update an existing comment on an issue or pull request.

Useful for workflows that post status summaries on PRs (docs review, test results, deployment status) and need to update a single comment instead of spamming new ones on every push.

**Component:** `github.updateIssueComment`
- Config: repository, commentId, body
- API: `PATCH /repos/{owner}/{repo}/issues/comments/{comment_id}`
- Output: updated comment object

**Usage pattern:**
1. First run: `createIssueComment` -> store comment ID in memory
2. Subsequent runs: read ID from memory -> `updateIssueComment`

PR: #5349

### PR DESCRIPTION
**Pull request by bender-rodriguez-unit1**
## What

New GitHub action component to edit an existing comment on an issue or PR.

Closes #5361

## Why

Workflows that post status summaries on PRs (docs review, test results, deployment status) need to update a single comment instead of spamming new ones on every push.

## Component

**`github.updateIssueComment`**
- **Config:** repository, commentId, body
- **Output:** updated comment object (id, url, body, timestamps)
- **API:** `PATCH /repos/{owner}/{repo}/issues/comments/{comment_id}`

## Usage Pattern

1. First run: `createIssueComment` -> store comment ID in memory
2. Subsequent runs: read ID from memory -> `updateIssueComment`

## Changes

- `update_issue_comment.go` — new component with validated parseCommentID (handles scientific notation, NaN, decimals, range)
- `update_issue_comment_test.go` — Setup + Execute tests (11 cases)
- `client.go` — added `EditIssueComment` wrapper
- `example.go` — embedded example output
- `payloads/update_issue_comment.json` — example payload matching GitHub API schema
- `github.go` — registered component
- `capability_mapper.go` — registered in PermissionIssues group
- `GitHub.mdx` — auto-generated docs updated
- `update_issue_comment.ts` — frontend mapper
- `update_issue_comment.spec.ts` — frontend mapper test
- `index.ts` — frontend registration

### CODE REVIEW

File: pkg/integrations/github/components/issues/payloads/update_issue_comment.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "url": "https://api.github.com/repos/superplanehq/superplane/issues/comments/1234567890",
+  "html_url": "https://github.com/superplanehq/superplane/issues/42#issuecomment-1234567890",
+  "body": "## Docs Review Summary\n\n**Verdict:** No update needed\n\nAll changes are internal.",
+  "user": {
+    "login": "superplane-bot",
+    "id": 12345678
+  },
+  "created_at": "2026-06-11T12:00:00Z",
+  "updated_at": "2026-06-11T14:30:00Z"
+}
```
Comment (shiroyasha) [inline pkg/integrations/github/components/issues/payloads/update_issue_comment.json:37]: This is a suspiciously short example payload.

**Comment by shiroyasha** [comment]
Everything looks good. But please check if the example output is real.

File: pkg/integrations/github/github.go [hunk 2]
```diff
@@ -121,6 +121,7 @@ func (g *GitHub) Actions() []core.Action {
 		&issues.AddIssueLabel{},
 		&issues.CreateIssue{},
 		&issues.CreateIssueComment{},
+		&issues.UpdateIssueComment{},
```
[the remark this answers is not in the document]
Comment (bender-rodriguez-unit1) [reply pkg/integrations/github/github.go:124]: Already fixed in ee2c492 — added to `PermissionIssues` group in capability mapper.

File: pkg/integrations/github/components/issues/update_issue_comment.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	updatedComment, _, err := client.EditIssueComment(context.Background(), config.Repository, commentID, comment)
+	if err != nil {
+		return fmt.Errorf("failed to update issue comment: %w", err)
+	}
+
+	return ctx.ExecutionState.Emit(
+		core.DefaultOutputChannel.Name,
+		"github.issueComment.updated",
+		[]any{updatedComment},
+	)
```
[the remark this answers is not in the document]
Comment (bender-rodriguez-unit1) [reply pkg/integrations/github/components/issues/update_issue_comment.go:167]: Valid. Fixed in fc185d9 — changed to `github.issueComment` to match the convention.

File: pkg/integrations/github/components/issues/payloads/update_issue_comment.json (hunk 1 above)
Comment (bender-rodriguez-unit1) [reply pkg/integrations/github/components/issues/payloads/update_issue_comment.json:37]: Fixed in 6769b2a — now matches the `createIssueComment` format with full data/timestamp/type wrapper.

File: pkg/integrations/github/components/issues/update_issue_comment.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+func (c *UpdateIssueComment) Execute(ctx core.ExecutionContext) error {
+	var config UpdateIssueCommentConfiguration
+	if err := mapstructure.Decode(ctx.Configuration, &config); err != nil {
+		return fmt.Errorf("failed to decode configuration: %w", err)
+	}
+
+	commentID, err := strconv.ParseInt(config.CommentID, 10, 64)
+	if err != nil {
+		return fmt.Errorf("comment ID is not a valid number: %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (bender-rodriguez-unit1) [reply pkg/integrations/github/components/issues/update_issue_comment.go:147]: Valid. Fixed in 6769b2a — added `parseCommentID` that tries `ParseInt` first, falls back to `ParseFloat` for scientific notation. Tests cover both formats.

File: pkg/integrations/github/components/issues/update_issue_comment.go [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+	// Try integer first
+	if id, err := strconv.ParseInt(s, 10, 64); err == nil {
+		return id, nil
+	}
+	// Fall back to float parsing for scientific notation
+	f, err := strconv.ParseFloat(s, 64)
+	if err != nil {
+		return 0, err
+	}
+	return int64(f), nil
```
[the remark this answers is not in the document]
Comment (bender-rodriguez-unit1) [reply pkg/integrations/github/components/issues/update_issue_comment.go:209]: Valid. Fixed in 8783913 — replicated the full validated parser from `pulls/add_reaction.go` with NaN, Inf, decimal, and range checks. Tests cover all edge cases.

File: pkg/integrations/github/components/issues/update_issue_comment.go [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+		return fmt.Errorf("failed to decode configuration: %w", err)
+	}
+
+	if config.Repository == "" {
+		return errors.New("repository is required")
+	}
+
+	if config.CommentID == "" {
+		return errors.New("comment ID is required")
+	}
```
[the remark this answers is not in the document]
Comment (bender-rodriguez-unit1) [reply pkg/integrations/github/components/issues/update_issue_comment.go:120]: Valid. Fixed in 869956a — Setup now trims all config fields with `strings.TrimSpace` so whitespace-only values are caught at publish time. Added test.
